nameko/messaging.py

```diff
@@ -350,8 +350,11 @@ def on_iteration(self):
             self.should_stop = True
 
     def on_connection_error(self, exc, interval):
-        _log.warn('broker connection error: {}. '
-                  'Retrying in {} seconds.'.format(exc, interval))
+        _log.warn(
+            "Error connecting to broker at {} ({}).\n"
+            "Is the broker running? "
+            "Do you have permission to access to the vhost?\n"
+            "Retrying in {} seconds.".format(self.amqp_uri, exc, interval))
```

Comment:
rather that doing this here (i.e. on all connection errors), maybe we could do a connection-check, e.g. in `setup`

it doesn't look like rabbit is of much help (e.g. bad creds result in a pause when connecting, before a disconnect)

pika tries to track how far it gets during connection and gives heuristic error messages, but amqp doesn't help very much. best i could do was something like this

```
from amqp import Connection

class ConnectionTest(Connection):
    def __init__(self, *a, **k):
        try:
            super(ConnectionTest, self).__init__(*a, **k)
        except IOError as exc:
            if not hasattr(self, '_wait_tune_ok'):
                raise
            elif self._wait_tune_ok:
                print "probably bad creds"
            else:
                print "probably bad vhost"
```